tests/upgrade/test_upgrade_from.sh

```diff
@@ -0,0 +1,228 @@
+#!/bin/bash
+
+# This test checks for control and data plane upgrade from the passed in version to $HUB/$TAG.
+# It runs the following steps:
+# 1. Installs istio with multiple gateway replicas at source version. Version is defined by tag e.g. 1.0.0
+# 2. Installs fortio echosrv with a couple of different subsets/destination rules with multiple replicas.
+# 3. Sends external traffic to echosrv through ingress gateway.
+# 4. Sends internal traffic to echosrv from fortio load pod.
+# 5. Upgrades control plane to current head version.
```

Comment:
Is it really upgrading to HEAD? It seems that the upgrade relies on HUB/TAG, which would have to be pushed in a separate step.  This allows room for user error if they forget to push images for the current build. We should probably call that out here?

Reply:
This comment was out of date - thanks for catching that.